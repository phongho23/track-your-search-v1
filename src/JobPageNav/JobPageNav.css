.JobPageNav__back-button-wrapper {
    max-width: 380px;
    margin: 0 10px;
  }
  
  .JobPageNav__back-button {
    margin: 0 auto 20px;
  }
  
  .JobPageNav__back-button:hover {
    background-color: #505256;
  }
  
  .JobPageNav__week-name {
  color: white;
  font-family: var(--custom-font);
  padding-left: 5%;
  padding-right: 10%;
  padding-top: 5px;
  display: flex;
  padding: 5px 5px;
  font-size: 32px;
  }