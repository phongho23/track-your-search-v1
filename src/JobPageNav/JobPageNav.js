
import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import CircleButton from '../CircleButton/CircleButton'
import ApiContext from '../ApiContext'
import { findJob, findWeek } from '../jobs-helpers'
import './JobPageNav.css'


export default class JobPageNav extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { }
    },
    match: {
      params: {}
    }
  }
  static contextType = ApiContext;

  render() {

    const { jobs, weeks, } = this.context
    const { jobId } = this.props.match.params
    const job = findJob(jobs, jobId) || {}
    const week = findWeek(weeks, job.weekId)

  return (
    <div className='JobPageNav'>
      <div className='JobPageNav__back-button-wrapper'>
      <CircleButton
          tag='button'
          role='link'
          type='button'
          onClick={() => this.props.history.goBack()}
          className='JobPageNav__back-button'
      >
          <FontAwesomeIcon icon='chevron-left' />
          <br />
          Back
      </CircleButton>
      </div>
      {week && (
        <h3 className='JobPageNav__week-name'>
          {week.name}
        </h3>
      )}
    </div>
    )
  }
}


