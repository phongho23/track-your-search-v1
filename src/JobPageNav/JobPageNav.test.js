import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JobPageNav from './JobPageNav'

describe(`JobPageNav component`, () => {
  it('renders a .JobPageNav by default', () => {
    const wrapper = shallow(<JobPageNav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it.skip('renders a h3 with week name when in props', () => {
    const props = {
      match: {
        params: {
          jobId: 'test-job-id'
        }
      }
    }
    const context = {
      jobs: [{ id: 'test-job-id', weekId: 'test-week-id' }],
      weeks: [{ id: 'test-week-id', name: 'Important' }]
    }

    const h3 = shallow(<JobPageNav {...props} />, context)
      .find('.JobPageNav__week-name')
    expect(toJson(h3)).toMatchSnapshot()
  })
})
