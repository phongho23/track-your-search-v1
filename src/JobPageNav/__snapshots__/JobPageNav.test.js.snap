// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPageNav component renders a .JobPageNav by default 1`] = `
<div
  className="JobPageNav"
>
  <div
    className="JobPageNav__back-button-wrapper"
  >
    <NavCircleButton
      className="JobPageNav__back-button"
      onClick={[Function]}
      role="link"
      tag="button"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="chevron-left"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Back
    </NavCircleButton>
  </div>
</div>
`;
